mempool/mempool.go

```diff
@@ -875,15 +880,15 @@ func (mp *TxPool) ProcessOrphans(hash *chainhash.Hash) []*btcutil.Tx {
 //
 // This function is safe for concurrent access.
 func (mp *TxPool) ProcessTransaction(tx *btcutil.Tx, allowOrphan, rateLimit bool) ([]*btcutil.Tx, error) {
+	log.Tracef("Processing transaction %v", tx.Hash())
+
 	// Protect concurrent access.
 	mp.mtx.Lock()
-	defer mp.mtx.Unlock()
```

Comment:
I'd prefer to leave this one.  When it's a simple function with only a single control path, I think it makes sense to avoid to the defer.  However, in a more complex function like this with multiple control paths, it's too easy to add a branch and forget to release the mutex.